pages/[id].js

```diff
@@ -36,30 +36,33 @@ export default function PlantDetails() {
 
   return (
     <>
-      <Link href="/">Back</Link>
-
-      <Container>
-        <h1>Details Page</h1>
+      <Link href="/" aria-label="Back to Home">
+        Back
+      </Link>
+      <Container aria-labelledby="details-heading">
+        <h1 id="details-heading">Details Page</h1>
         <Image
-          alt={`image of ${plant.name}`}
+          alt={`Image of ${plant.name}`}
           src={plant.imageUrl}
           width={200}
           height={200}
         />
-        <h2>{plant.name}</h2>
-        <p>{plant.botanicalName}</p>
-        <p>{plant.description}</p>
-        <p>
+        <h2 aria-label={`Plant name: ${plant.name}`}>{plant.name}</h2>
+        <p aria-label={`Botanical name: ${plant.botanicalName}`}>
+          {plant.botanicalName}
+        </p>
+        <p aria-label={`Description of ${plant.name}`}> {plant.description} </p>
+        <p aria-label={`Light need for ${plant.name}`}>
           Light: {lightNeedIcon[plant.lightNeed] || "🌞"} {plant.lightNeed}
         </p>
-        <p>
+        <p aria-label={`Water need for ${plant.name}`}>
           Water need: {waterNeedIcon[plant.waterNeed] || "🌱"} {plant.waterNeed}
         </p>
-        <p>
+        <p aria-label="Fertiliser season">
           Fertiliser season:
           {plant.fertiliserSeason.map((season) => (
-            <span key={season}>
-              {seasonIcons[season] || "🌱"} {season}{" "}
+            <span key={season} aria-label={`Season: ${season}`}>
+              {seasonIcons[season] || "🌱"} {season}
             </span>
           ))}
         </p>
```

Comment:
I recommend to create a PlantDetails component and render it here. You can copy and paste this code to the component. The reason is separating of concerns. This file (`[id].js`) is for rendering the page e.g. with some standard layout but for rendering the deails info it's a good practice to have a reusable component.